src/test/java/com/example/demo/F1DriverStatsControllerTest.java

```diff
@@ -0,0 +1,242 @@
+package com.example.demo;
+
+import com.example.demo.controller.F1DriverStatsController;
+import com.example.demo.model.Driver;
+import com.example.demo.model.DriverRepository;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+//@SpringBootTest
+@WebMvcTest(F1DriverStatsController.class)
+public class F1DriverStatsControllerTest {
+    @Autowired
+    private MockMvc mockMvc;
+    @MockBean
+    private DriverRepository driverRepository;
+
+    @Test
+    public void getDriverByID() throws Exception {
+        doReturn(Optional.of(
+                new Driver(1L, "Michael", "Schumacher", 91, 68, 70, 2500)))
+                .when(driverRepository).findById(anyLong());
+
+        mockMvc.perform(
+                MockMvcRequestBuilders.get("/f1stats/1")
+        )
+                .andExpect(status()
+                        .isOk())
+                .andExpect(content()
+                        .json("{\"id\":1,\"firstName\":Michael,\"lastName\":Schumacher,\"wins\":91,\"poles\":68,\"fastestLaps\":70,\"points\":2500.0}"));
```

Comment:
Avoid hard-coded values like this JSON string. All tests would fail if we just add a field to the Driver class, right? You should construct the expected JSON at runtime using some library like Gson, for example.